devito/types/grid.py

```diff
@@ -739,12 +742,15 @@ def __subdomain_finalize__(self, grid, **kwargs):
             # equivalent.
             self._local_bounds = self._global_bounds
 
-    def _create_implicit_exprs(self, grid):
+    @cached_property
+    def _implicit_exprs(self):
```

Comment:
for another day: we could probably detach the generation logic from this class and move it inside the compiler pass directly, it'd be neater probably